src/Client/ClientBase.cpp

```diff
@@ -1474,6 +1475,23 @@ void ClientBase::processInsertQuery(const String & query_to_execute, ASTPtr pars
     }
 }
 
+namespace
+{
+    bool isStdinNotEmptyAndValid(ReadBufferFromFileDescriptor & std_in)
```

Comment:
Maybe we should add the check here as well 
https://github.com/ClickHouse/ClickHouse/blob/a9960a146535b976a69939f53190a1d941a4f698/src/Client/ClientBase.cpp#L1438  ?